controllers/admin.js

```diff
@@ -315,11 +315,22 @@ exports.getOrderDetails = async (req, res, next) => {
     const order = await Order.findById(req.params.orderId);
     const riders = await Rider.find();
 
+    let orderAssignedTo = null;
+
+    for (let i in riders) {
```

Comment:
This part here is too slow. A better way would be to add an 'assignedTo' field in the order schema itself and add a rider ref. What do you say?